### ISSUE TITLE
upstream check in alpine version

### ISSUE THREAD

**Issue by vpusher** [nginxinc/docker-nginx#173]
I'm a running the `1.13.1-alpine` version of the docker image in front of a tomcat. 

In my NGinx configuration I have a simple proxy pass like this one:

```
proxy_pass http://tomcat:8080/app/; 
```

The `tomcat` hostname is resolved by the DNS Service provided by [Rancher](http://rancher.com/). Each containers (Nginx and Tomcat) are created through Rancher with a compose files.

The issue here is that Nginx can't start since he cannot resolve the `tomcat` upstream, ending up with this error:

```
... host not found in upstream "tomcat"
```

Nevertheless, using the non-alpine image (`1.13.1`), it works like a charm. NGinx manages to start.

So what's the difference in the resolution flow between alpine and non-alpine version of the image ? Is it a bug in the alpine one ? Do I miss something ?

**Comment by thresheek** [comment]
Alpine has a history of DNS resolving bugs, see https://github.com/nginxinc/docker-nginx/issues/78 e.g. - can you test with FROM alpine:3.6 possibly?

**Comment by jmagnusson** [comment]
@thresheek I just did a custom build of [this](https://github.com/nginxinc/docker-nginx/blob/849fed0093112cd9f55491fccd2f861eb9fad5f9/stable/alpine/Dockerfile) with `FROM` replaced with `alpine:3.6`, but the problem persists.

I can confirm that the problem exists in `1.12.1-alpine` and `1.13.3-alpine` as well.

I'm doing local development with docker-compose spinning up the proxy backend and nginx. If the backend starts after nginx this problem comes up. So the problem doesn't seem to be Rancher specific.

The best solution I've found so far is to use [wait-for-it](https://github.com/vishnubob/wait-for-it) in a Dockerfile like this:

```Dockerfile
FROM nginx:1.13.3-alpine

RUN apk add --no-cache bash
COPY wait-for-it.sh /wait-for-it.sh
RUN chmod +x /wait-for-it.sh

CMD ["/wait-for-it.sh", "app:5011", "--", "nginx", "-g", "daemon off;"]
```
